src/WebJobs.Script/HostOptionsProvider.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Azure.WebJobs.Host.Config;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Azure.WebJobs.Hosting;
+using Microsoft.Azure.WebJobs.Script.Extensions;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Serialization;
+
+namespace Microsoft.Azure.WebJobs.Script
+{
+    public class HostOptionsProvider : IHostOptionsProvider
+    {
+        private readonly JsonSerializer _serializer = JsonSerializer.Create(new JsonSerializerSettings
+        {
+            ContractResolver = new CamelCasePropertyNamesContractResolver(),
+        });
+
+        private readonly IEnumerable<IExtensionOptionsProvider> _extensionOptionsInfoCollection;
+        private readonly ILogger<HostOptionsProvider> _logger;
+        private readonly IOptionsMonitor<ConcurrencyOptions> _concurrencyOptions;
+
+        public HostOptionsProvider(IEnumerable<IExtensionOptionsProvider> extensionOptionsInfoCollection, IOptionsMonitor<ConcurrencyOptions> concurrencyOption, ILogger<HostOptionsProvider> logger)
+        {
+            _extensionOptionsInfoCollection = extensionOptionsInfoCollection;
+            _logger = logger;
+            _concurrencyOptions = concurrencyOption;
+        }
+
+        public JObject GetOptions()
+        {
+            var payload = new JObject();
+            var extensions = new JObject();
+            var extensionOptions = GetExtensionOptions();
+            foreach (var extension in extensionOptions)
+            {
+                extensions.Add(extension.Key, extension.Value);
+            }
+            if (extensionOptions.Count != 0)
+            {
+                payload.Add("extensions", extensions);
+            }
+
+            var concurrency = GetConcurrencyOptions();
+            payload.Add("concurrency", concurrency);
+            return payload;
+        }
+
+        private Dictionary<string, JObject> GetExtensionOptions()
+        {
+            Dictionary<string, JObject> result = new Dictionary<string, JObject>();
+            foreach (IExtensionOptionsProvider extensionOptionsInfo in _extensionOptionsInfoCollection)
```

Comment:
nit: rename extensionOptionsProvider and rename collection as noted above

Reply:
Done